pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go

```diff
@@ -237,3 +243,259 @@ func TestDeleteAddressWithWrongTier(t *testing.T) {
 		})
 	}
 }
+
+const (
+	gceProjectId = "test-project"
+	gceRegion    = "us-central1"
+	zoneName     = "us-central1-b"
+	nodeName     = "test-node-1"
+	clusterName  = "Test Cluster Name"
+	clusterID    = "test-cluster-id"
+)
+
+var apiService = &v1.Service{
+	Spec: v1.ServiceSpec{
+		SessionAffinity: v1.ServiceAffinityClientIP,
+		Type:            v1.ServiceTypeClusterIP,
+		Ports:           []v1.ServicePort{{Protocol: v1.ProtocolTCP, Port: int32(123)}},
+	},
+}
+
+func fakeGCECloud() (*GCECloud, error) {
+	client, err := newOauthClient(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	service, err := compute.New(client)
+	if err != nil {
+		return nil, err
+	}
+
+	// Used in disk unit tests
+	fakeManager := newFakeManager(gceProjectId, gceRegion)
+	zonesWithNodes := createNodeZones([]string{zoneName})
+
+	alphaFeatureGate, err := NewAlphaFeatureGate([]string{})
+	if err != nil {
+		return nil, err
+	}
+
+	cloud := cloud.NewMockGCE()
+	cloud.MockTargetPools.AddInstanceHook = mock.AddInstanceHook
+	cloud.MockTargetPools.RemoveInstanceHook = mock.RemoveInstanceHook
+
+	gce := GCECloud{
+		region:             gceRegion,
+		service:            service,
+		manager:            fakeManager,
+		managedZones:       []string{zoneName},
+		projectID:          gceProjectId,
+		AlphaFeatureGate:   alphaFeatureGate,
+		nodeZones:          zonesWithNodes,
+		nodeInformerSynced: func() bool { return true },
+		c:                  cloud,
+	}
+
+	return &gce, nil
+}
+
+func createAndInsertNodes(gce *GCECloud, nodeNames []string) ([]*v1.Node, error) {
+	nodes := []*v1.Node{}
+
+	for _, name := range nodeNames {
+		// Inserting the same node name twice causes an error - here we call
+		// DeleteInstance to guarantee that nodes aren't inserted twice.
+		// See TestUpdateExternalLoadBalancer.
+		err := gce.DeleteInstance(gceProjectId, zoneName, name)
+
+		err = gce.InsertInstance(
+			gceProjectId,
+			zoneName,
+			&compute.Instance{
+				Name: name,
+				Tags: &compute.Tags{
+					Items: []string{name},
+				},
+			},
+		)
+
+		if err != nil {
```

Comment:
I see that in many places you put a newline before the err check. We block them together.